client/disputes/evidence/index.js

```diff
@@ -353,252 +47,85 @@ const getDisputeProductType = ( dispute ) => {
-			: __( 'Evidence saved!', 'woocommerce-payments' );
-		const href = addQueryArgs( 'admin.php', {
-			page: 'wc-admin',
-			path: '/payments/disputes',
+	const doRemoveFile = ( key ) => {
+		updateEvidence( key, '' );
+		updateDispute( {
+			...dispute,
+			metadata: { ...dispute.metadata, [ key ]: '' },
+			fileSize: { ...dispute.fileSize, [ key ]: 0 },
 		} );
-
-		wcpayTracks.recordEvent(
-			submit
-				? 'wcpay_dispute_submit_evidence_success'
-				: 'wcpay_dispute_save_evidence_success'
-		);
-		/*
-			We rely on WC-Admin Transient notices to display success message.
-			https://github.com/woocommerce/woocommerce-admin/tree/master/client/layout/transient-notices.
-		*/
-		createSuccessNotice( message, {
-			actions: [
-				{
-					label: submit
-						? __(
-								'View submitted evidence',
-								'woocommerce-payments'
-						  )
-						: __(
-								'Return to evidence submission',
-								'woocommerce-payments'
-						  ),
-					url: addQueryArgs( 'admin.php', {
-						page: 'wc-admin',
-						path: '/payments/disputes/challenge',
-						id: query.id,
-					} ),
-				},
-			],
+		updateEvidenceUploadErrorsForDispute( disputeId, {
+			...evidenceUploadErrors,
+			[ key ]: '',
 		} );
-
-		getHistory().push( href );
 	};
 
-	const handleSaveError = ( err, submit ) => {
-		wcpayTracks.recordEvent(
-			submit
-				? 'wcpay_dispute_submit_evidence_failed'
-				: 'wcpay_dispute_save_evidence_failed'
-		);
-
-		const message = submit
-			? __( 'Failed to submit evidence. (%s)', 'woocommerce-payments' )
-			: __( 'Failed to save evidence. (%s)', 'woocommerce-payments' );
-		createErrorNotice( sprintf( message, err.message ) );
+	const doUploadFile = async ( key, file ) => {
+		uploadFileEvidenceForDispute( disputeId, key, file );
 	};
 
-	const { updateDispute: updateDisputeInStore } = useDisputeEvidence();
-
 	const doSave = async ( submit ) => {
-		// Prevent submit if upload is in progress.
-		if ( isUploadingEvidence() ) {
-			createInfoNotice(
-				__(
-					'Please wait until file upload is finished',
-					'woocommerce-payments'
-				)
-			);
-			return;
-		}
-
-		setLoading( true );
-
-		try {
-			wcpayTracks.recordEvent(
-				submit
-					? 'wcpay_dispute_submit_evidence_clicked'
-					: 'wcpay_dispute_save_evidence_clicked'
-			);
-
-			const { metadata } = dispute;
-			const updatedDispute = await apiFetch( {
-				path,
-				method: 'post',
-				data: {
-					// Send full evidence, as submission does not appear to work without new evidence despite being optional.
-					evidence: { ...dispute.evidence, ...evidence },
-					metadata,
-					submit,
-				},
-			} );
-			setDispute( updatedDispute );
-			handleSaveSuccess( submit );
-			setEvidence( {} );
-			updateDisputeInStore( updatedDispute );
-		} catch ( err ) {
-			handleSaveError( err, submit );
-		} finally {
-			setLoading( false );
+		if ( submit ) {
+			submitEvidence( dispute.id, evidenceTransient );
+		} else {
+			saveEvidence( dispute.id, evidenceTransient );
 		}
```

Comment:
Currently we're ignoring the return value from `submitEvidence` and `saveEvidence`, so maybe we don't need to return anything? It may be useful to keep it in case we want to move some of the notice functionality out of the actions in the data layer and into the component.